src/graph/GoExecutor.cpp

```diff
@@ -375,14 +381,18 @@ std::vector<VertexID> GoExecutor::getDstIdsFromResp(RpcResponse &rpcResp) const
     return std::vector<VertexID>(set.begin(), set.end());
 }
 
-
 void GoExecutor::finishExecution(RpcResponse &&rpcResp) {
-    auto outputs = setupInterimResult(std::move(rpcResp));
+    std::unique_ptr<InterimResult> outputs;
+
+    if (!setupInterimResult(std::move(rpcResp), outputs)) {
+        return;
```

Comment:
do onError_()

Reply:
Not needed here, because returning false in setupInterimResult (processFinalResult ) means that onError_ has already been called.